File: backend/app/models/mcp_integration.py
"""MCP integration request/response models."""

from __future__ import annotations

from pydantic import BaseModel, Field


class McpIntegrationSettings(BaseModel):
    enabled: bool = False
    server_url: str = ""
    api_key: str | None = Field(
        default=None,
        description="Set to update API key; omit or null to keep existing value.",
    )
    use_kubernetes: bool = True
    use_aws: bool = True
    use_cloud_cost: bool = True
    use_pr_reviewer: bool = True


class McpIntegrationResponse(BaseModel):
    enabled: bool
    server_url: str
    api_key_configured: bool
    api_key_preview: str | None = None
    use_kubernetes: bool
    use_aws: bool
    use_cloud_cost: bool
    use_pr_reviewer: bool
    instance_server_configured: bool


class McpTestResponse(BaseModel):
    status: str
    message: str
    tool_count: int = 0
    resource_count: int = 0
    tools: list[str] = Field(default_factory=list)


class McpAskRequest(BaseModel):
    question: str = Field(min_length=1, max_length=4000)


class McpToolCallRecord(BaseModel):
    tool_name: str
    arguments: dict = Field(default_factory=dict)
    result_summary: str = ""


class McpAskResponse(BaseModel):
    answer: str
    tools_used: list[McpToolCallRecord] = Field(default_factory=list)

File: frontend/types/mcpIntegration.ts
export interface McpIntegrationSettings {
  enabled: boolean;
  server_url: string;
  api_key?: string | null;
  use_kubernetes: boolean;
  use_aws: boolean;
  use_cloud_cost: boolean;
  use_pr_reviewer: boolean;
}

export interface McpIntegrationResponse {
  enabled: boolean;
  server_url: string;
  api_key_configured: boolean;
  api_key_preview: string | null;
  use_kubernetes: boolean;
  use_aws: boolean;
  use_cloud_cost: boolean;
  use_pr_reviewer: boolean;
  instance_server_configured: boolean;
}

export interface McpTestResponse {
  status: string;
  message: string;
  tool_count: number;
  resource_count: number;
  tools: string[];
}

export interface McpToolCallRecord {
  tool_name: string;
  arguments: Record<string, unknown>;
  result_summary: string;
}

export interface McpAskRequest {
  question: string;
}

export interface McpAskResponse {
  answer: string;
  tools_used: McpToolCallRecord[];
}
